crates/biome_js_analyze/src/lint/nursery/no_useless_length_check.rs

```diff
@@ -0,0 +1,364 @@
+                ErrorType::UselessLengthCheckWithSome => T![&&],
+            };
+            if logical_exp.operator_token().ok()?.kind() != operator {
+                return None;
+            };
+            let left = logical_exp.left().ok()?;
+            let left_replacer = (logical_exp.clone(), logical_exp.right().ok()?, left.range());
+            search_logical_exp(
+                &left,
+                Some(left_replacer),
+                expect_error,
+                comparing_zeros,
+                array_tokens_used_api,
+            )?;
+
+            let right = logical_exp.right().ok()?;
+            let right_replacer = (logical_exp.clone(), logical_exp.left().ok()?, right.range());
+            search_logical_exp(
+                &right,
+                Some(right_replacer),
+                expect_error,
+                comparing_zeros,
+                array_tokens_used_api,
+            )
+        }
+        // a === 0 ext.
+        AnyJsExpression::JsBinaryExpression(binary_exp) => {
+            let comparing_zero = get_comparing_length_exp(binary_exp, expect_error)?;
+            let AnyJsExpression::JsIdentifierExpression(array_token) = comparing_zero else {
+                return None;
+            };
+            let key = array_token.text();
+            if let Some(comparing_zero_list) = comparing_zeros.get_mut(&key) {
+                comparing_zero_list.push(replacer?);
+            } else {
+                comparing_zeros.insert(key, vec![replacer?]);
+            }
+            Some(())
+        }
+        // .some() or .every() etc.
+        AnyJsExpression::JsCallExpression(task_exp) => {
+            if task_exp.is_optional_chain() {
+                return None;
+            }
+            let task_member_exp =
+                AnyJsMemberExpression::cast(task_exp.callee().ok()?.into_syntax())?;
+            let task_target = task_member_exp.object().ok()?;
+            let AnyJsExpression::JsIdentifierExpression(task_target_token) = task_target else {
+                return None;
+            };
+
+            let task_member_name_node = task_member_exp.member_name()?;
+            let task_member_name = task_member_name_node.text();
+            match expect_error {
+                ErrorType::UselessLengthCheckWithEvery => {
+                    if task_member_name == "every" {
+                        array_tokens_used_api.insert(task_target_token.text());
+                        Some(())
+                    } else {
+                        None
+                    }
+                }
+                ErrorType::UselessLengthCheckWithSome => {
+                    if task_member_name == "some" {
+                        array_tokens_used_api.insert(task_target_token.text());
+                        Some(())
+                    } else {
+                        None
+                    }
+                }
+            }
+        }
+        // ( hoge )
```

Comment:
Actually, hoge is only used in Japanese community as foo😅
```suggestion
        // ( foo )
```